cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs

```diff
@@ -78,56 +80,101 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
-  H.diffFileVsGoldenFile
-    protocolParametersOutFile
-    "test/cardano-testnet-test/files/golden/queries/protocolParametersFileOut.json"
-
-  -- tip to stdout
-  _ :: QueryTipLocalStateOutput <- H.noteShowM $ H.execCliStdoutToJson execConfig [ "conway", "query", "tip" ]
-  -- tip to a file
-  let tipOutFile = work </> "tip-out.json"
-  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "tip"
-                                   , "--out-file", tipOutFile]
-  _ :: QueryTipLocalStateOutput <- H.leftFailM . H.readJsonFile $ tipOutFile
-
-  -- stake-pools to stdout
-  stakePoolsOut <- H.execCli' execConfig [ "conway", "query", "stake-pools" ]
-  length (lines stakePoolsOut) H.=== 3 -- Because, by default, 3 stake pools are created
-  -- Light test of the query's answer, the ids should exist:
-  forM_ (lines stakePoolsOut) $ \stakePoolId -> do
-    H.execCli' execConfig [ "conway", "query", "pool-state"
-                          , "--stake-pool-id", stakePoolId ]
-  -- stake-pools to a file
-  let stakePoolsOutFile = work </> "stake-pools-out.json"
-  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "stake-pools"
-                                   , "--out-file", stakePoolsOutFile]
-
-  -- query drep-state to stdout
-  -- TODO: deserialize to a Haskell value when
-  -- https://github.com/IntersectMBO/cardano-cli/issues/606 is tackled
-  dreps :: Aeson.Value <- H.noteShowM $ H.execCliStdoutToJson execConfig [ "conway", "query", "drep-state", "--all-dreps"]
-  assertArrayOfSize dreps 3
-  -- query drep-state to a file
-  let drepStateOutFile = work </> "drep-state-out.json"
-  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "drep-state", "--all-dreps"
-                                   , "--out-file", drepStateOutFile]
-  _ :: Aeson.Value <- H.leftFailM . H.readJsonFile $ drepStateOutFile
+
+  -- protocol-parameters
+  do
+    -- to stdout
+    protocolParametersOut <- H.execCli' execConfig [ eraName, "query", "protocol-parameters" ]
+    H.diffVsGoldenFile
+      protocolParametersOut
+      "test/cardano-testnet-test/files/golden/queries/protocolParametersOut.txt"
+    -- protocol-parameters to a file
+    let protocolParametersOutFile = work </> "protocol-parameters-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "protocol-parameters"
+                                     , "--out-file", protocolParametersOutFile]
+    H.diffFileVsGoldenFile
+      protocolParametersOutFile
+      "test/cardano-testnet-test/files/golden/queries/protocolParametersFileOut.json"
+
+  -- tip
+  do
+    -- to stdout
+    _ :: QueryTipLocalStateOutput <- H.noteShowM $ H.execCliStdoutToJson execConfig [ eraName, "query", "tip" ]
+    -- to a file
+    let tipOutFile = work </> "tip-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "tip"
+                                     , "--out-file", tipOutFile]
+    _ :: QueryTipLocalStateOutput <- H.readJsonFileOk tipOutFile
```

Comment:
Should we use a bang pattern here to be sure we are actually decoding the file?

Reply:
That should be done by `readJsonFileOk` which lifts the `Either DecodeError QueryTipLocalStateOutput` to `MonadTest m => m QueryTipLocalStateOutput`, so it should error out at this place.